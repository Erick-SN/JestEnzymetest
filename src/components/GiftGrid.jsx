import React from 'react';
import PropTypes from 'prop-types';
import GiftGridItem from './GiftGridItem';
import { useFetch } from '../hooks/useFetch';

const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetch(category);
  return (
    <>
      <h3 className='animate_animated animate_fadeIn'>{category}</h3>
      {loading && <p className='animate_animated animate_flash'>Loading...</p>}
      <div className='card-grid'>
        {images.map((img) => (
          <GiftGridItem key={img.id} {...img}>
            img.title
          </GiftGridItem>
        ))}
      </div>
    </>
  );
};

GiftGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GiftGrid;
