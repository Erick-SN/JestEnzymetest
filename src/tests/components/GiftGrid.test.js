import React from 'react';
import GiftGrid from '../../components/GiftGrid';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { useFetch } from '../../hooks/useFetch';
jest.mock('../../hooks/useFetch');

describe('Test at <GiftGrid/>', () => {
  const category = 'One Punch';
  test('Render <GiftGrid/> component', () => {
    useFetch.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('shoud load images', () => {
    const imgs = [
      {
        id: 'abc',
        url: 'https://cosa.jpg',
        title: 'image',
      },
    ];

    useFetch.mockReturnValue({
      data: imgs,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GiftGridItem').length).toBe(imgs.length);
  });
});
