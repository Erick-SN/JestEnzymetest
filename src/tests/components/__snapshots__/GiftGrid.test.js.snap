// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GiftGrid/> Render <GiftGrid/> component 1`] = `ShallowWrapper {}`;

exports[`Test at <GiftGrid/> shoud load images 1`] = `ShallowWrapper {}`;
